src/hotspot/share/oops/method.cpp

```diff
@@ -566,16 +566,23 @@ MethodCounters* Method::build_method_counters(Thread* current, Method* m) {
   methodHandle mh(current, m);
   MethodCounters* counters;
   if (current->is_Java_thread()) {
-    // For when TRAPS is JavaThread.
-    counters = MethodCounters::allocate(mh, current->as_Java_thread());
+    Thread* THREAD = current;
+    counters = MethodCounters::allocate(mh, THREAD);
```

Comment:
Can you add a comment before this line:

// Use the TRAPS version for a JavaThread so it will adjust the GC  threshold if needed.

Thanks.

Reply:
ok, yes that says why.